client/lib/signup/step-actions.js

```diff
@@ -512,6 +512,29 @@ export function createAccount(
 
 				// Fire after a new user registers.
 				analytics.recordRegistration();
+				/**
+				 * Auto login the user when it has been created.
+				 */
+				user.clear();
+				user.fetching = false;
+
+				user.on( 'change', () => {
+					const newLoggedUser = user.get();
+
+					if ( newLoggedUser && newLoggedUser.ID ) {
+						analytics.identifyUser( newLoggedUser.ID, newLoggedUser.username );
+					} else {
+						/**
+						 * The user fetching is a bit fragile and sometimes it requires to do a double-fetch.
```

Comment:
@jaswrks Thanks for checking in on this code. 🙇 

I really haven't had time to go deep on this. I reverted #11715 in full, and [the comment first appeared there](https://github.com/Automattic/wp-calypso/pull/11715/files#diff-160d380c201b4db2d947c495e877a98aR286), so hopefully @bisko can answer this when he's back from AFK.

Reply:
When I originally worked on that there was some kind of a caching issue in the `User` object that had to be "cleared" and fetched twice to get the latest data.
I'm not sure if it was a backend or a front end issue though.

It's possible things have changed since then with all the latest reworks towards a better data layer.

Reply:
Thanks! Maybe what we can do is add a logger to catch when this occurs and monitor for a while to see if we can safely remove this then. Notes on arbitrary error logging in Calypso [here](https://github.com/Automattic/wp-calypso/tree/master/client/lib/catch-js-errors#arbitrary-logging).

Reply:
Those errors get sent to Kibana and are found using `feature:calypso`.

Reply:
Thanks a lot for the help on this one @jaswrks and @bisko 

I've added a `log()` call as you suggested